src/request.ts

```diff
@@ -14,25 +14,69 @@ export interface RequestOptions {
   tokenProvider?: TokenProvider;
 }
 
+export interface RawRequestOptions {
+  method: string;
+  url: string;
+  jwt?: string;
+  headers?: ElementsHeaders;
+  body?: any;
+  logger?: Logger;
+}
+
 // TODO: Could we make this generic and remove the `any`s?
 export function executeNetworkRequest(
   createXhr: () => XMLHttpRequest,
   options: RequestOptions,
 ): Promise<any> {
   return new Promise<any>((resolve, reject) => {
-    const xhr = createXhr();
-
-    xhr.onreadystatechange = () => {
-      if (xhr.readyState === 4) {
-        if (xhr.status >= 200 && xhr.status < 300) {
-          resolve(xhr.response);
-        } else if (xhr.status !== 0) {
-          reject(ErrorResponse.fromXHR(xhr));
-        } else {
-          reject(new NetworkError('No Connection'));
+    const xhr = attachOnReadyStateChangeHandler(createXhr(), resolve, reject);
+
+    xhr.send(JSON.stringify(options.body));
+  });
+}
+
+export function executeRawNetworkRequest(
```

Comment:
Can we use a shorter name like `sendRawRequest`?

Reply:
Also, do we need `executeNetworkRequest` anymore?

Reply:
Will shorten the name 👍 

We do still need `executeNetworkRequest`, unfortunately, because it has some special behaviour, namely: 

- automatically `JSON.stringify`s any `body` that's provided
- automatically sets the `Content-Type` header to `application/json` if a `body` is present
- has a different API for getting an `xhr` object to use: i.e. ends up calling this https://github.com/pusher/pusher-platform-js/blob/master/src/transport/http.ts#L327-L347